File: java/src/WhereUsedResponder.java

class WhereUsedResponder extends ResultResponder {
	public String title() {
		return "Where Used: " + search_for_page();
	}

	public boolean traverse(WikiPage page) {
		return page.text().contains(search_for_page());
	}

	private String search_for_page() {
		return request.data().get("where_used");
	}
}

File: csharp/WhereUsedResponder.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;

namespace WikiSearch
{
	public class WhereUsedResponder : ResultResponder
	{
		public override string Title()
		{
			return "Where Used: " + SearchForPage();
		}

		public override bool Traverse(WikiPage page)
		{
			return page.Text.Contains(SearchForPage());
		}

		private string SearchForPage()
		{
			return request.Data["where_used"];
		}
	}
}

